src/web/components/elements/InteractiveContentBlockElement.tsx

```diff
@@ -0,0 +1,272 @@
+
+	display: none;
+	${showStickyNavOption && 'display: grid;'}
+`;
+
+const wrapperStyles = css`
+	margin-top: -5px; /* a hack to get tighter margin between interactive element and Divider */
+
+	${from.tablet} {
+		margin-left: -20px;
+		width: calc(100% + 20px);
+	}
+	${from.leftCol} {
+		margin-left: -10px;
+		width: calc(100% + 10px);
+	}
+`;
+
+const stickyNavCurrentHeaderStyles = css`
+	height: ${space[12]}px;
+	padding: ${space[4]}px;
+
+	display: flex;
+	align-items: center;
+	${headline.xxsmall({ fontWeight: 'bold' })}
+
+	${until.mobileLandscape} {
+		${headline.xxxsmall({ fontWeight: 'bold' })}
+	}
+	position: fixed;
+
+	cursor: pointer;
+	border-top: 0;
+	box-shadow: 0 4px 12px 0 rgb(0 0 0 / 5%);
+`;
+
+const SVGStyles = css`
+	width: ${space[5]}px;
+	display: flex;
+	margin-right: ${space[2]}px;
+	transform: translateY(-3px) rotate(0deg);
+	transition: transform 250ms ease-out;
+`;
+
+const SVGTransitionStyles = css`
+	transform: translateY(-3px) rotate(180deg);
+	transition: transform 250ms ease-out;
+`;
+
+type Props = {
+	subheadingLinks: SubheadingBlockElement[];
+};
+
+interface EnhancedSubheadingType extends SubheadingBlockElement {
+	ref?: HTMLElement | null;
+}
+
+export const InteractiveContentBlockElement = ({ subheadingLinks }: Props) => {
+	const [showStickyNavOption, setShowStickyNavOption] = useState<boolean>(
+		false,
+	);
+
+	// Loop through subheadingLinks and add a reference + reverse each object
+	const [enhancedSubheadings, setEnhancedSubheadings] = useState<
+		EnhancedSubheadingType[]
+	>([]);
+	useEffect(() => {
+		setEnhancedSubheadings(
+			subheadingLinks
+				.map((subheadingLink) => {
+					return {
+						...subheadingLink,
+						ref: document.getElementById(subheadingLink.elementId),
+					};
+				})
+				// we reverse the list to make it easier to find the first element lowest down the page
+				// the first match will always be the one lowest down the page
+				.reverse(),
+		);
+	}, [subheadingLinks, setEnhancedSubheadings]);
+
+	// set the height explicitly of the container as to make sure that when we detach
+	// list elements we do not effect the page height
+	const [height, setHeight] = useState<number>();
+	const divRef = useCallback((node: HTMLDivElement | null) => {
+		if (node) setHeight(node?.getBoundingClientRect().height);
+	}, []);
+
+	// The sticky header needs to be based on the section of the article that is being view
+	// We accomplish this by
+	// -> determine coordinates of each subheadingLink
+	// -> check current page scroll using `window.scrollY`
+	// -> set sticky Nav object based on scroll
+	const [
+		stickyNavCurrentHeader,
+		setStickyNavCurrentHeader,
+	] = useState<null | EnhancedSubheadingType>(null);
+	useEffect(() => {
+		const onScroll = libDebounce(() => {
+			const firstElementTop = enhancedSubheadings[
+				enhancedSubheadings.length - 1
+			]?.ref?.getBoundingClientRect().top;
+
+			const articleBodyBottom = document
+				.getElementById('article-body')
```

Comment:
Is there another way to have a reference to this element, or even another way to calculate the position at which we should hide the sticky nav? Just feels like this id could be used/abused by other interactives creating implicit contracts.

Reply:
We should defo be using intersection observer